tests/feedback.test.ts

```diff
@@ -0,0 +1,147 @@
+
+    expect(submittedFeedback1).toHaveLength(1);
+    expect(submittedFeedback2).toHaveLength(1);
+    expect(submittedFeedback1[0]).toMatchObject(feedback1);
+    expect(submittedFeedback2[0]).toMatchObject(feedback2);
+    expect(allSubmittedFeedback).toHaveLength(2);
+    expect(allSubmittedFeedback).toEqual(expect.arrayContaining([...submittedFeedback1, ...submittedFeedback2]));
+  });
+
+  test('can be acknowledged and rewarded points by admin', async () => {
+    const [user] = UserFactory.create(1);
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [feedback] = FeedbackFactory.create(1);
+
+    await FeedbackControllerMock.mock([user, admin]);
+    const submittedFeedback = await FeedbackControllerMock.submitFeedback(feedback, user);
+    const acknowledgedFeedback = await FeedbackControllerMock.acknowledgeFeedback(submittedFeedback.uuid, admin);
+
+    const conn = await DatabaseConnection.get();
+    const persistedUser = await conn.manager.findOne(UserModel, { where: { uuid: user.uuid } });
+    const feedbackPointReward = Config.pointReward.FEEDBACK_POINT_REWARD;
+
+    expect(acknowledgedFeedback.status).toEqual(FeedbackStatus.ACKNOWLEDGED);
+    expect(persistedUser.points).toEqual(user.points + feedbackPointReward);
+  });
+
+  test('can be ignored and not rewarded anything by admin', async () => {
+    const [user] = UserFactory.create(1);
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [feedback] = FeedbackFactory.create(1);
+
+    await FeedbackControllerMock.mock([user, admin]);
+    const submittedFeedback = await FeedbackControllerMock.submitFeedback(feedback, user);
+    const ignoredFeedback = await FeedbackControllerMock.ignoreFeedback(submittedFeedback.uuid, admin);
+
+    const conn = await DatabaseConnection.get();
+    const persistedUser = await conn.manager.findOne(UserModel, { where: { uuid: user.uuid } });
+
+    expect(ignoredFeedback.status).toEqual(FeedbackStatus.IGNORED);
+    expect(persistedUser.points).toEqual(user.points);
+  });
+
+  test('cannot be ackowledged or ignored after already being acknowledged or ignored', async () => {
+    const [user] = UserFactory.create(1);
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [feedback1, feedback2] = FeedbackFactory.create(2);
+
+    await FeedbackControllerMock.mock([user, admin]);
+    const submittedFeedback1 = await FeedbackControllerMock.submitFeedback(feedback1, user);
+    const submittedFeedback2 = await FeedbackControllerMock.submitFeedback(feedback2, user);
+
+    const uuid1 = submittedFeedback1.uuid;
+    const uuid2 = submittedFeedback2.uuid;
+
+    await FeedbackControllerMock.acknowledgeFeedback(uuid1, admin);
+    await FeedbackControllerMock.ignoreFeedback(uuid2, admin);
+
+    const acknowledgeAcknowledgedFeedback = () => FeedbackControllerMock.acknowledgeFeedback(uuid1, admin);
```

Comment:
Why create single-use functions here and then call them right after? vs inlining them.

Reply:
Mainly for conveying via. function name what the function is doing, but with the variable renaming change above, these functions aren't needed anymore.